skimage/filters/tests/test_correlate.py

```diff
@@ -0,0 +1,21 @@
+import numpy as np
+from skimage._shared import testing
+from skimage._shared.testing import assert_equal, assert_almost_equal
+from skimage._shared.testing import assert_array_equal
+from scipy import ndimage as ndi
+from skimage.filters import correlate_sparse
+
+def test_correlate_sparse():
```

Comment:
@zeroth could you use `skimage._shared.testing.parametrize` (example use [here](https://github.com/scikit-image/scikit-image/blob/bab6e3b92596bdf020c5a402000112b39579b06e/skimage/filters/tests/test_window.py#L17-L20)) to test all available ndimage.correlate modes? *and* additionally, test `valid` separately since that requires different handling.

Finally, you should also use `testing.raises` to test what happens when you give an even-sized kernel.